### PR TITLE
new function in class map

### LINKED ISSUE #6: Zadanie 6 (getIsland)
W klasie `Map` napisz funkcję

`Island* getIsland(const Island::Coordinates& coordinate)`

Powinna ona przeszukać `std::vector<Island>` i zwrócić szukaną wyspę.

### PR DESCRIPTION
**Pull request by KGebski0036**

### CODE REVIEW

File: shm/inc/map.hpp [hunk 1]
```diff
@@ -2,10 +2,14 @@
...
+#include "coordinates.hpp"
 #include "island.hpp"
 
 class Map {
 private:
     std::vector<Island> islands_;
     Island* currentPosition_;
+
+public:
+    Island* getIsland(const Coordinates&);
```
Comment (arafalski) [inline shm/inc/map.hpp]: Could you add that section above private one?

File: shm/src/map.cpp [hunk 2]
```diff
@@ -1 +1,9 @@
 #include "map.hpp"
+
+#include <algorithm>
+Island* Map::getIsland(const Coordinates& coordinate) {
+    return &*(std::find_if(islands_.begin(), islands_.end(),
+                           [&](auto el) {
```
Comment (arafalski) [inline shm/src/map.cpp]: I think that this parameter should be passed by const reference.

**Review by arafalski — CHANGES_REQUESTED** [review]

File: shm/inc/map.hpp (hunk 1 above)
Comment (kontrybutor) [inline shm/inc/map.hpp]: also, Mateusz said that's good to add name of the parameter in function declaration:
```suggestion
    Island* getIsland(const Coordinates& coordinates);
```

File: shm/src/map.cpp [hunk 3]
```diff
@@ -1 +1,9 @@
 #include "map.hpp"
+
+#include <algorithm>
+Island* Map::getIsland(const Coordinates& coordinate) {
```
Comment (kontrybutor) [inline shm/src/map.cpp:4]: I would add one line after `#include`

**Review by arafalski — DISMISSED** [review]

File: shm/src/map.cpp [hunk 4]
```diff
@@ -1 +1,10 @@
 #include "map.hpp"
+
+#include <algorithm>
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    return &*(std::find_if(islands_.begin(), islands_.end(),
+                           [&](const auto& el) {
+                               return coordinate == el.getPosition();
+                           }));
+}
```
Comment (kontrybutor) [inline shm/src/map.cpp]: Shouldn't we return `nullptr` if island is not found?

```suggestion

    auto island = std::find_if(islands_.begin(), islands_.end(),
                           [&](const auto& island) {
                               return coordinate == island.getPosition();
                           }));
    return island != std::end(islands_) ? &*island : nullptr;
}
```
I've changed also name of lambda parameter. I think '`island` is better. What do you think?

File: shm/inc/map.hpp [hunk 5]
```diff
@@ -2,9 +2,13 @@
 
 #include <vector>
 
+#include "coordinates.hpp"
 #include "island.hpp"
 
 class Map {
+public:
+    Island* getIsland(const Coordinates& coordinate);
```
Comment (kontrybutor) [inline shm/inc/map.hpp]:
```suggestion
#include "island.hpp"

class Map {
public:
    Island* getIsland(const Coordinates& coordinates);
```
As `coordinate` is a verb, `coordinates` is a noun :)

File: shm/src/map.cpp (hunk 4 above)
Comment (KGebski0036) [reply shm/src/map.cpp]: Thanks, I forgot about this

**Review by arafalski — APPROVED** [review]

**Review by kontrybutor — APPROVED** [review]

**Pull request merged**